com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/TimeProfiler.java

```diff
@@ -255,21 +216,20 @@ public static void createReport(Boolean fromEditor) {
                     writer.flush();
                     IOUtils.closeQuietly(writer);
                 }
-                catch (IOException e) {
-                    throw e;
-                }
             }
         } catch (Exception e) {
             throw new RuntimeException(e);
+        } finally {
+            marks.clear();
+            currentScopes.clear();
         }
 
         long reportEndTime = time();
-        System.out.printf("\nTime profiler report creation took %.2f seconds", (reportEndTime - reportStartTime)/1000.0f);
+        System.out.printf("Time profiler report creation took %.2f seconds\n", (reportEndTime - reportStartTime)/1000.0f);
     }
 
     private static long getCurrentThreadId() {
-        Thread t = Thread.currentThread();
-        return (t != null) ? t.getId() : MAIN_THREAD_ID;
+        return Thread.currentThread().getId();
```

Comment:
`Thread.currentThread()` is never null.

Reply:
Oh, nice, I assumed it could be null for some reason...